library-best-practices/JLBP-10.md

```diff
@@ -0,0 +1,33 @@
+[JLBP-10] Infrequent breaking changes
+-------------------------------------
+
+Every breaking change incurs a cost to the consumers. There is the immediate
+cost of the consumers who are forced to change their own code to adapt to the
+newer version, but the far higher cost is the inconsistency introduced into the
+ecosystem and the resulting potential for diamond dependency conflicts. The
+deeper a library is in the dependency tree, the higher the inconsistency cost,
+regardless of the magnitude of the direct change cost.
+
+Consequently, a library should try to make breaking changes very infrequently if
+ever. It's hard to recommend a frequency for all cases because of the widely
+varying cost. On one end of the spectrum are pre-1.0 libraries which don't
+promise stability, and which can have breaking changes between every release. On
+the other end are Java standard libraries which all libraries depend on, most of
+whose classes have not had any breaking changes since they were introduced. As
+an anchoring point, no more than once every five years might be a good default
```

Comment:
I don't think a specific duration matters a great deal. What's more important is that the current version without breaking changes is supported for 5+ years or until no one's using it. A new branch can move forward at any time.

Reply:
It doesn't feel like you're necessarily disagreeing with the text; the text suggests five years "as an anchoring point," but still leaves the exact time frame up to the library maintainer.